### DISCUSSION TITLE
[FR] Add event to fire "before save field"

### DISCUSSION THREAD

**Discussion by lindseydiloreto** [craftcms/cms#12362] [Ideas] [answered]
**TL;DR** - This already exists!! (sorry) 🤦‍♂️

---

Humbly requesting a new Event to be added **before the field settings are saved**. This would allow me to sort out some nuances in the field settings that are otherwise difficult or impossible to address.

### Real-World Example

In my plugin's field settings, there is a **nested array**. Because of the way that HTML is structured, this data must be submitted via a front-end form in the following format...

```html
<input type="checkbox" name="types[namespace\MyField][nestedArray][0][enabled]" value="1">
```

The problem is, this stores the data as the string "1" in both the `projectconfig` table and in the YAML files.

This is not ideal. I would much rather be able to save that value as `true`, which I know that the `projectconfig` table and YAML files can handle. Technically I can provide a few workarounds for "truthy" values, but I'd much rather be storing those as boolean values to begin with.

### Conclusion

If there were an event triggered by the `fields/save-field` controller action, I could use it to typecast my array before that data is saved to the `projectconfig` table and the YAML files.

**Comment by lindseydiloreto** [comment, marked as answer]
I stand corrected, this Event already exists! Just had to dig a little deeper I guess...

```php
use yii\base\Event;
use craft\base\Field;
use craft\events\ModelEvent;

Event::on(
    Field::class,
    Field::EVENT_BEFORE_SAVE,
    function (ModelEvent $event) {
        // ...
    }
);
```

Sorry for the red herring, you can disregard this entire thread! 🙏
